### PR TITLE
FIX: Default state was null and could need to crashes (race conditions)

### PR DESCRIPTION
**Pull request by marcosrdz**

### CODE REVIEW

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @Overtorment] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/Overtorment

**Comment by limpbrains** [comment]
I have doubts about this solution. What if the function you are calling is a critical one, such as setting a password or saving a wallet? If it merely executes a stub function instead of performing its intended task, you could lose important data.

Additionally, if the stub function returns an empty string, the app may crash because it cannot parse it.

I don’t think this is a good approach. It would be better to address the specific issue causing the failure. Perhaps consider delaying the app's rendering while this variable is being initialized.

**Comment by marcosrdz** [comment]
> I have doubts about this solution. What if the function you are calling is a critical one, such as setting a password or saving a wallet? If it merely executes a stub function instead of performing its intended task, you could lose important data.
> 
> 
> 
> Additionally, if the stub function returns an empty string, the app may crash because it cannot parse it.
> 
> 
> 
> I don’t think this is a good approach. It would be better to address the specific issue causing the failure. Perhaps consider delaying the app's rendering while this variable is being initialized.

True. This makes me think perhaps those critical functions shouldn't even by inside the Provider.

**Comment by limpbrains** [comment]
I would say anything that doesn't need react context doesn't need to inside the Provider

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @limpbrains] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/limpbrains

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @Overtorment] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/Overtorment

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @limpbrains] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/limpbrains

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @Overtorment] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/Overtorment

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @limpbrains] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/limpbrains

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @Overtorment] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/Overtorment

File: blue_modules/storage-context.tsx [hunk 1]
```diff
@@ -60,8 +60,48 @@ export enum WalletTransactionsStatus {
...
+  encryptStorage: async () => {},
+  sleep: async () => {},
+  createFakeStorage: async _password => false,
+  decryptStorage: async (_password: string) => false,
+  isPasswordInUse: async () => false,
+  cachedPassword: '',
+  getItem: async () => '',
+  setItem: async () => {},
+};
+
```
Comment (Overtorment) [inline blue_modules/storage-context.tsx:103]: is there a better way then to provide a dummy state with all the fields thats not gona be actually used? feels like a waste, and like smart people probably figured out a solution

**Review by Overtorment — CHANGES_REQUESTED** [review]
tests failed

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @limpbrains] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/limpbrains

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @limpbrains] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/limpbrains

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/h2dvuldij6yo5txy2pu2iilaim?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/1885)

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @Overtorment] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/Overtorment

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @limpbrains] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/limpbrains

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @Overtorment] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/Overtorment

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @limpbrains] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/limpbrains

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @Overtorment] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/Overtorment

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @limpbrains] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/limpbrains

**Comment by limpbrains** [comment]
Looks like new solution is just the same as old, with a few functions from BlueApp, right? 
So the issue remains - dummy functions that are not doing anything.
If this is the only solution, maybe add logging or throw an error inside the function, so we can know when dummy is being called?

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @Overtorment] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/Overtorment

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @limpbrains] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/limpbrains

**Review by Overtorment — APPROVED** [review]

File: components/Context/StorageProvider.tsx [hunk 2]
```diff
@@ -59,25 +59,65 @@ export enum WalletTransactionsStatus {
...
+const encryptStorage = BlueApp.encryptStorage;
+const sleep = BlueApp.sleep;
+const createFakeStorage = BlueApp.createFakeStorage;
+const decryptStorage = BlueApp.decryptStorage;
+const isPasswordInUse = BlueApp.isPasswordInUse;
+const cachedPassword = BlueApp.cachedPassword;
+const getItem = BlueApp.getItem;
+const setItem = BlueApp.setItem;
+
+export const StorageContext = createContext<StorageContextType>(defaultContext);
```
Comment (Overtorment) [inline components/Context/StorageProvider.tsx:117]: not ts expert, but is it possible to throw away  `StorageContextType`
and just use `typeof defaultContext`, if we are going with using `defaultContext` anyway?

Comment (marcosrdz) [reply components/Context/StorageProvider.tsx:117]: defaultContext is an object. StorageContextType is the type

Comment (Overtorment) [reply components/Context/StorageProvider.tsx:117]: yeah so construction like this, it returns type? not sure, im bad at ts

`typeof defaultContext`

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @limpbrains] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/limpbrains

**Comment by Overtorment** [comment]
ok i dont really like the solution here. there must be a better way

**Comment by limpbrains** [comment]
Could it be beause of some parts are now lazy-loaded?

**Comment by marcosrdz** [comment]
> Could it be beause of some parts are now lazy-loaded?

Only the screens past WalletsList. Everything else is the same

**Comment by limpbrains** [comment]
How do I replicate this issue?
I'm thinking, maybe only render `children` in StorageProvider only if everything is ready?
Something like
```jsx
<StorageContext.Provider value={value}>{value !== undefined && children}</StorageContext.Provider>
```

**Pull request closed without merging**